ST/36_Streams & lambda expressions/Streams/StreamSolutions.java

```diff
@@ -0,0 +1,98 @@
+import java.util.*;
+import java.util.stream.*;
+import java.util.function.Predicate;
+
+public class StreamSolutions {
+
+    public static void main(String[] args) {
+        // Running each solution method
+        solutionForQuestion1();
+        solutionForQuestion2();
+        solutionForQuestion3();
+        solutionForQuestion4();
+        solutionForQuestion5();
+    }
+
+    // Solution for Question 1
+    // Given a list of transactions, how would you use streams to find the average amount of all transactions?
+    private static void solutionForQuestion1() {
+        // Creating a list to hold transactions. (In a real-world application, this list would be populated from a data source)
+        List<Transaction> transactions = new ArrayList<>();
+        double averageAmount = transactions.stream()
+                // Mapping each transaction to its amount
+                .mapToDouble(Transaction::getAmount)
+                .average()
+                // Providing a default value if there are no transactions
+                .orElse(0.0);
```

Comment:
can we also add an example with .orElseThrow()?

Reply:
Done.

I added an alternative solution, where in case calculating the average is not possible due to missing values, an exception is thrown.